### PR TITLE
core: new-infra: merge `TrackObject` and `Detector` objects

### PR DESCRIPTION
**Pull request by eckter**

### CODE REVIEW

File: core/src/main/java/fr/sncf/osrd/new_infra/implementation/GraphHelpers.java [hunk 1]
```diff
@@ -39,6 +40,31 @@ public static TrackNode getCommonNode(ImmutableNetwork<TrackNode, TrackEdge> und
         return intersection.stream().iterator().next();
     }
 
+    /** Returns true if the edges share a node */
+    public static boolean areEdgesAdjacent(ImmutableNetwork<TrackNode, TrackEdge> undirectedGraph,
```
Comment (flomonster) [inline core/src/main/java/fr/sncf/osrd/new_infra/implementation/GraphHelpers.java]: We could test these utility functions to satisfy Codecov

**Review by flomonster — DISMISSED** [review]
Great!

**Review by multun — DISMISSED** [review]

**Comment by multun** [comment]
I've noticed you've added unused helpers, do you plan to use these soon?

File: core/src/main/java/fr/sncf/osrd/new_infra/implementation/GraphHelpers.java (hunk 1 above)
Comment (eckter) [reply core/src/main/java/fr/sncf/osrd/new_infra/implementation/GraphHelpers.java]: This was supposed to go in another separate commit, I'll rewrite some history

**Comment by eckter** [comment]
> I've noticed you've added unused helpers, do you plan to use these soon?

Yes, in https://github.com/DGEXSolutions/osrd/pull/725/files actually, but it was an error to include those in this commit. I've moved them .

**Review by multun — APPROVED** [review]

**Pull request merged**